rooms.js

```diff
@@ -163,6 +165,95 @@ var Room = (function () {
 		}
 		return true;
 	};
+	//mute handling
+	Room.prototype.runMuteTimer = function () {
+		if (this.muteTimer) return;
+		if (this.mutedUsers[0].time < Date.now() + 5) {
+			// this prevents lag from bugging the timer
+			this.mutedUsers[0].time = Date.now() + 5;
+		}
+		var self = this;
+		this.muteTimer = setTimeout(function () {
+			var user = Users(self.mutedUsers[0].userid);
```

Comment:
actually on a related note, is `if (user)` ever actually needed? I just realized the logic there is bad because the timer will never clear if `user` is undefined. I haven't had that happen--even when an offline user became unmuted it still worked. Is there any case where a `user` could end up undefined?